evaluation.md

```diff
@@ -6,14 +6,136 @@ _Describe each change from your ideal syntax to the syntax you implemented, and
 describe_ why _you made the change._
 
 **On a scale of 1–10 (where 10 is "a lot"), how much did you have to change your syntax?**
+The following are some observations we made and challenges we encountered when creating our internal DSL.
+
+* Initially, we thought it would be very easy to move sequentially from "command" to the next (each line of instructions is a command). However, we then realized that it's tricky to have some commands include an explicit designation of the next command to jump to, and for other commands to not include that explicit designation. At first, we didn't want to write " +1 " at the end of every command if we wanted the machine to proceed by evaluating the next step. We only wanted to specify the next step if we wanted the machine to jump to any command NOT directly after the current command, check if the condition of the command is satisfied, move in the designated direction if so, and check the condition of the next command if the condition of the current command is not satisfied. We also wanted to use the asterisk to instruct the machine to reevaluate the condition of the current command (the condition being the combination of free/occupied States) again, and move accordingly, until the condition of that command is no logner satisfied.. However, we didn't know how to keep track of a counter that knew the current state, so we had to explicitly name the CurrentState and the NextState.  
+
+* Throughout writing the internal DSL, we realized that it was very clever for the original Piconot language to present the state of the 4 RelativeDescriptions using an asterisk, N/E/W/S, or an X. This is really convenient from both the parsing perspective, and for conveniently translating from user input to create a new "Surrroundings" object. Our method for representing whether or not a RelativeDescription could be anything, occupied, or free, seemed like a good idea at first, but then proved to be difficult. Our language asks the user to put the letters of free RelativeDescriptions inside (), and to put the letters of free RelativeDescriptions inside []. The user does not have to explicitly name the letters that could be Anything (either free or occupied). However, this posed some challenges for creating the DSL.
```

Comment:
What we did was we had the Surroundings object be filled with `Anything` for each position, and then based on the directions the user gave as blocked or free, we would change the corresponding direction to `Blocked` or `Free`.  I think this worked pretty well for us as a workaround to this problem.